### ISSUE TITLE
Constructible Permalink to a Course

### ISSUE THREAD

**Issue by hanzhi713** [thecourseforum/theCourseForum2#353]
Hi, I'm the maintainer of [plannable](https://github.com/awesome-schedule/plannable), an online course scheduling website for UVa. 

I would like to add a link (a button) on our course information modal to the course review page (https://thecourseforum.com/course/xxxx). A brief inspection of the source code suggests that xxxx is the primary key of the course.  To be able to make the link, I need to find a way to get this pk, which I currently have no idea. 

I know Lou's list has links to theCourseForum and is able to match the pks. I'm wondering how this is achieved. My hypothesis is that the course pks start from 0 and are autoincremented when new (unique) courses are added. 

To sum up,
1. I would like to know if there is any way to obtain a course's pk from its relevant information. 
2. If it is not possible, can you provide an alternative permalink constructible from its relevant information. 

The relevant information here refers to semester name (e.g. Fall 2020), course department (e.g. CS), course number (e.g. 1110), course type (e.g. Lecture/Lab/Discussion/etc.), and possibly some other attributes of the course.

**Comment by hanzhi713** [comment]
Might relate to #333

**Comment by j-alicia-long** [comment]
Hello @hanzhi713, thanks so much for reaching out! We'd love to collaborate with you on this. 

I can provide you with a db dump of the course pks for now, and once #333 gets completed we'll update you if any changes are needed. Could you please email info@thecourseforum.com so we can reply with the data?

**Comment by j-alicia-long** [comment]
Also, we'd love to feature a link to Plannable on our site since we no longer have our own scheduler. Is this something you are open to?

**Comment by hanzhi713** [comment]
> Hello @hanzhi713, thanks so much for reaching out! We'd love to collaborate with you on this.
> 
> I can provide you with a db dump of the course pks for now, and once #333 gets completed we'll update you if any changes are needed. Could you please email [info@thecourseforum.com](mailto:info@thecourseforum.com) so we can reply with the data?

I think I'll just wait until #333 is finalized and merged because incorporating a database dump is more error-prone.

**Comment by hanzhi713** [comment]
> Also, we'd love to feature a link to Plannable on our site since we no longer have our own scheduler. Is this something you are open to?

We'd love to! You can just put a link to https://plannable.org somewhere on your website.

**Comment by hanzhi713** [comment]
Plannable v9.3 has been released, and links to theCourseForum review page are included in this new release. Release note: https://github.com/awesome-schedule/plannable/releases/tag/v9.3.0 

@j-alicia-long Thank you and your team for merging #333. It will be awesome if you can add a link to https://plannable.org somewhere on your website.
